src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c

```diff
@@ -41,29 +33,267 @@ const ActorInit Arrow_Fire_InitVars = {
+        ArrowFire_SetupAction(this, FireArrow_Hit);
+        this->timer = 32;
+        this->alpha = 255;
+        return;
+    }
+    if (arrow->unk_260 < 34) {
+        if (this->alpha < 35) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        this->alpha -= 25;
+    }
+}
+
+void ArrowFire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ArrowFire* this = (ArrowFire*)thisx;
+
+    if ((globalCtx->msgCtx.msgMode == 0xE) || (globalCtx->msgCtx.msgMode == 0x12)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->actionFunc(this, globalCtx);
+}
+
+void FireArrow_SetQuadVerticies(ArrowFire* this) {
+    static Vec3f D_80922284 = { 100.0f, 700.0f, 0.0f };
+    static Vec3f D_80922290 = { 400.0f, 27.0f, 0.0f };
+    static Vec3f D_8092229C = { -100.0f, 700.0f, 0.0f };
+    static Vec3f D_809222A8 = { -400, 27.0f, 0.0f };
+    static Vec3f D_809222B4 = { 0.0f, 700.0f, 100.0f };
+    static Vec3f D_809222C0 = { 0.0f, 27.0f, 400.0f };
+    static Vec3f D_809222CC = { 0.0f, 700.0f, -100 };
+    static Vec3f D_809222D8 = { 0.0f, 27.0f, -400 };
+    Vec3f sp44;
+    Vec3f sp38;
+    Vec3f sp2C;
+    Vec3f sp20;
+
+    Matrix_MultiplyVector3fByState(&D_80922284, &sp44);
+    Matrix_MultiplyVector3fByState(&D_80922290, &sp38);
+    Matrix_MultiplyVector3fByState(&D_8092229C, &sp2C);
+    Matrix_MultiplyVector3fByState(&D_809222A8, &sp20);
+    Collider_SetQuadVertices(&this->collider1, &sp44, &sp38, &sp2C, &sp20);
+    Matrix_MultiplyVector3fByState(&D_809222B4, &sp44);
+    Matrix_MultiplyVector3fByState(&D_809222C0, &sp38);
+    Matrix_MultiplyVector3fByState(&D_809222CC, &sp2C);
+    Matrix_MultiplyVector3fByState(&D_809222D8, &sp20);
+    Collider_SetQuadVertices(&this->collider2, &sp44, &sp38, &sp2C, &sp20);
+}
+
+void ArrowFire_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnArrow* arrow;
+    ArrowFire* this = THIS;
+    u32 frames = globalCtx->state.frames;
+    s32 pad;
+
+    arrow = (EnArrow*)this->actor.parent;
+
+    if ((arrow != NULL) && (arrow->actor.update != NULL) && (this->timer < 255)) {
+        Actor* transform = (arrow->unk_261 & 2) ? &this->actor : &arrow->actor;
+
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        Matrix_InsertTranslation(transform->world.pos.x, transform->world.pos.y, transform->world.pos.z, MTXMODE_NEW);
+        Matrix_RotateY(transform->shape.rot.y, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(transform->shape.rot.x, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(transform->shape.rot.z, MTXMODE_APPLY);
+
+        Matrix_Scale(0.01f, 0.01f, 0.01f, MTXMODE_APPLY);
+        if (this->blueingEffectMagnitude > 0.0f) {
+            POLY_XLU_DISP = func_8012BFC4(POLY_XLU_DISP);
+
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, (s32)(this->blueingEffectMagnitude * 40.0f) & 0xFF, 0, 0,
+                            (s32)(150.0f * this->blueingEffectMagnitude) & 0xFF);
```

Comment:
this tints the screen red, not sure if `blueingEffectMagnitude` makes sense. same for light arrows with making the screen yellow
naming it after screenFill makes more sense imo

Reply:
I just took it from ice arrows. I can change it there as well if it makes sense.

Reply:
yea its called blueing there cause it makes the screen blue. But yea it controls the color and the alpha of the screen fill, so screenFillIntensity makes sense to me personally